### PR TITLE
Deviceplugin refactoring: break Manager into DevicePluginHandler and endpoint

### LINKED ISSUE #51993: Device plugin code cleanup
/kind feature

Here are the possible code cleanup/refactoring we have discussed during device plugin PR code review or offline:
1. Currently Kubelet side device plugin is structured as three-layers: cm.DevicePluginHandler -> deviceplugin.Manager -> deviceplugin.endpoint. We may improve code readability by dropping them to two: cm.DevicePluginHandler -> deviceplugin.Endpoint. Some of the functionalities of deviceplugin.Manager will be moved to cm.DevicePluginHandler and some may be moved to deviceplugin.Endpoint.
2. The current deviceplugin.Endpoint has two methods: list and listAndWatch that make the same gRPC call. We may merge them into one. Combined with 1) item, the deviceplugin.Endpoint interface may expose the following methods: Start(), Stop(), Alloc(), run().
- Interface cleanup: some of the interfaces in deviceplugin pkg is not necessary public and can be changed to internal.
- Most functions in deviceplugin/utils.go doesn't seem to be necessary. May consider to remove this file.

### PR DESCRIPTION
**Pull request by lichuqiang**
**What this PR does / why we need it**:
From a design perspective, it seems clearer to keep Manager independent; However, people might feel it redundant when looking into the code, since the manager itself does not have much logic, largely works as passthrough, except for the gRPC server part.

**Which issue this PR fixes** 
 fixes #51993
Part3

**Special notes for your reviewer**:
/cc @jiayingz @RenaudWasTaken @vishh 

**Release note**:

`NONE`

### CODE REVIEW

File: pkg/kubelet/deviceplugin/manager.go [hunk 1]
```diff
@@ -1,236 +0,0 @@
...
-	}
-
-	return devs
-}
-
-// Allocate is the call that you can use to allocate a set of devices
-// from the registered device plugins.
-func (m *ManagerImpl) Allocate(resourceName string, devs []string) (*pluginapi.AllocateResponse, error) {
-
-	if len(devs) == 0 {
```
Comment (dashpole) [inline pkg/kubelet/deviceplugin/manager.go]: Looks like this check doesnt show up in DevicePluginHandlerImpl.Allocate.  Is it safe to drop this check?

**Review by dashpole — COMMENTED** [review]
Looks like it mostly just moves code around.

File: pkg/kubelet/deviceplugin/manager.go (hunk 1 above)
Comment (lichuqiang) [reply pkg/kubelet/deviceplugin/manager.go]: Should be, the allocation action of specific device plugins is supposed to be actions in a range of the devs, the return directly if devs empty.
Anyway, I would prefer add the check back for safety :)

**Comment by lichuqiang** [comment]
Close in favor of new implementation of #55180

**Pull request closed without merging**